lib/linguist/generated.rb

```diff
@@ -274,6 +275,18 @@ def generated_protocol_buffer?
       return lines[0].include?("Generated by the protocol buffer compiler.  DO NOT EDIT!")
     end
 
+    # Internal: Is the blob a Javascript source file generated by the
+    # Protocol Buffer compiler?
+    #
+    # Returns true of false.
+    def generated_javscript_protocol_buffer?
+      return false unless extname == ".js"
+      return false unless lines.count > 6
+
+      return lines[5].include?("GENERATED CODE -- DO NOT EDIT!")
```

Comment:
It seems to be hardcoded in the js generator right now: https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/js/js_generator.cc#L1383

But Javascript support is technically only Alpha right now, so it may change.